### ISSUE TITLE
@StoreRestResource for entity with multiple content properties

### ISSUE THREAD

**Issue by vierbergenlars** [paulcwarren/spring-content#1049]
**Is your feature request related to a problem? Please describe.**

I have a Contract class with multiple content properties.
Without customizations, these result in nice links and link relations based on their "prefix" names.
However, if I don't like those names, and I want to change them with `@StoreRestResource`, I run into the issue that all the content-properties are forced to the same relation (and path if I set that too).

```java
public class Contract {
	@Id
	private UUID id;

	@ContentId
	private UUID contentId;

	@ContentLength
	private long contentLength;


	@ContentId
	private UUID previewId;

	@ContentLength
	private long previewLength;
}

@StoreRestResource(linkRel = "test")
interface ContractContentStore extends ContentStore<Contract, UUID> {
}
```

```
$ http --json post :8889/contracts  number=1234
HTTP/1.1 201 
Connection: keep-alive
Content-Type: application/json
Date: Tue, 23 Aug 2022 15:32:05 GMT
Keep-Alive: timeout=60
Location: http://localhost:8889/contracts/302659cc-55a9-4786-b7a6-bc5ad857c524
Transfer-Encoding: chunked
Vary: Origin
Vary: Access-Control-Request-Method
Vary: Access-Control-Request-Headers

{
    "_links": {
        "contract": {
            "href": "http://localhost:8889/contracts/302659cc-55a9-4786-b7a6-bc5ad857c524"
        },
        "my-rel": [
            {
                "href": "http://localhost:8889/contracts/302659cc-55a9-4786-b7a6-bc5ad857c524/preview"
            },
            {
                "href": "http://localhost:8889/contracts/302659cc-55a9-4786-b7a6-bc5ad857c524/content"
            }
        ],
        "self": {
            "href": "http://localhost:8889/contracts/302659cc-55a9-4786-b7a6-bc5ad857c524"
        }
    },
    "contentId": null,
    "contentLength": 0,
    "number": "1234",
    "previewId": null,
    "previewLength": 0
}
```

**Describe the solution you'd like**

spring-data-rest allows having a `@RestResource()`-annotation on the *relation* field to change link relation and path.
Something similar on the @ContentId`-annotated property would enable us to solve this problem.

In case of "nested" content properties (with an embedded object), this could potentially even be done recursively. That would be helpful when in case we can have the same object class embedded multiple times (see also #1047), we can still get different complete paths.

**Describe alternatives you've considered**

**Additional context**

**Comment by paulcwarren** [comment]
Hi @vierbergenlars,

I liked the suggestion of using `@RestResource` but in the end decided, given prior [conversations](https://github.com/paulcwarren/spring-content/pull/1041) and the fact that SC walks entity class structures to generate a content property path/content property mapping context, it was simpler to suffix custom linkrel with the content property path as this is guaranteed to be unique (per store) and also works for nested properties.

I updated the docs to try and explain this as best I can.

If you have a reason why you need to customize the linkrel further - justifying @RestResource - then feel free to re-open this and I can look into adding that.

**Comment by paulcwarren** [comment]
Re-opening after conversation with @vierbergenlars.

**Comment by paulcwarren** [comment]
@vierbergenlars 

We would like to be able to influence the linkrel name that is used for a content property.  Sometimes a property (as defined on the entity) might be named something like `_package` (`package` being a Java keyword) and therefore the computed linkrel would also be `_package`.  But the User would like the linkrel to be called `package` as that is the system name of the content property and therefore how it is known by the system as a whole.

So, we have a `storerestresource` linkrel that allows User to customize top-level linkrel.  This doesn't work (as noted in the original bug report) when there are multiple content properties. 

Every contentproperty has a "path" - a /-separated path navigating from the root entity to the content property.  We can use this but this would also lead to a linkrel with `_package`; e.g. `primaryContent/_package`.  Not solving the problem.

Using `@RestResource` (as suggested by @vierbergenlars above) would allow us to influence this path.   We would need to construct a new path.  During which, at each level SCR would check the associated field to see if there is a `@RestResource` and if so replace the default name with its rel.

So, for the following Entity (trying to map a more complex scenario here):

```java
public class Contract {
	@Id
	private Long id;

        @Embedded
        private Content primary;

        @Embedded
        private Content secondary;
}

public class Content {
        @RestResource(rel="package")
	@ContentId
	private UUID _packageId;

	@ContentLength
	private long _packageLength;

        @RestResource(rel="private")
	@ContentId
	private UUID _privateId;

	@ContentLength
	private long _privateLength;
}
```
the computed linkrels would be:
-  `primary/package`
-  `primary/private`
-  `secondary/package`
-  `secondary/private`

If the store specified a linkRel:
```
@StoreRestResource(linkRel = "agreement")
interface ContractContentStore extends ContentStore<Contract, UUID> {
}
```

This would then be prefixed, as follows:
-  `agreement/primary/package`
-  `agreement/primary/private`
-  `agreement/secondary/package`
-  `agreement/secondary/private`

And in a simpler case:

```java
public class Contract {
	@Id
	private Long id;

        @RestResource(rel="package")
	@ContentId
	private UUID _packageId;

	@ContentLength
	private long _packageLength;

        @RestResource(rel="private")
	@ContentId
	private UUID _privateId;

	@ContentLength
	private long _privateLength;
}
```
the computed linkrels would be:
-  `package`
-  `private`

Satisfying the requirement, I believe.

**Comment by vierbergenlars** [comment]
To be honest, in the two cases that you describe, there is no real need to prefix "package" with an underscore. `package` is a reserved keyword and can't be used directly, but `packageId` or `packageLength` are fine.

My concrete case is having an embeddable "content"-object with only a single logical content property, which is then referenced from actual entities, like this:

```java
public class Contract {
  @Id
  private Long id;

  @Embedded
  @RestResource(rel="package")
  private Content _package;

  @Embedded
  @RestResource(rel="private")
  private Content _private;
}

public class Content {
  @ContentId
  private UUID id;

  @ContentLength
  private long length;
}
```

In this case, we want the resource relations on "contract" to be `package` and `private` respectively.

Thinking further ahead, there might be an even more complex nested structure. For JPA, this might not be a realistic scenario, but when using a document-database (e.g. mongo), objects can be more nested.

This is not a case that we currently actively need, but I think the naming follows logically from the descriptions above.

```java
public class Contract {
  @Id
  private Long id;

  @Embedded
  @RestResource(rel="package")
  private Content _package;

  @Embedded
  private Customer customer;
}

public class Customer {
  @Embedded
  @RestResource(rel="private")
  private Content _private;
}

public class Content {
  @ContentId
  private UUID id;

  @ContentLength
  private long length;
}
```

In that case, our content-relations on "contract" are expected to be `package` and `customer/private`.

Or, when setting an `@RestResource(rel="primary-customer")` on the `customer` property, that would result in a content-relation `primary-customer/private`.

I don't think an embedded object should ever get to dictate the full relation name. In the code example above, there would be no way for it to result in a content-relation named just `private`. (unless you want support having a specific `@RestResource(rel="")` omit that part of the relation name.)

**Comment by paulcwarren** [comment]
> To be honest, in the two cases that you describe, there is no real need to prefix "package" with an underscore. package is a reserved keyword and can't be used directly, but packageId or packageLength are fine.
Yeah...ahem...my bad...didnt think through that one!

All makes sense.

**Comment by paulcwarren** [comment]
Fixed with #1075
